### ISSUE TITLE
Can't manage to have a deviceID generated for android app

### ISSUE THREAD

**Issue by thomasgermain** [thomasloven/hass-browser_mod#222]
My Home Assistant version: 2021-05

**What I am doing:**
I'm trying to display a popup on my android app. This is working correctly on my chrome (on my laptop) browser. But I can't manage to make it works on my android app. Here is what I'm doing:
```
    tap_action:
      action: fire-dom-event
      browser_mod:
        command: popup
        deviceID:
          - this
        title: Alerte météo
        large: true
        card:
          type: markdown
          content: |
            {{'test'}}
```

Real problem is _I think_, that I'm not able to generate any deviceID for my android app. As far as I can understand, I just have to connect to HA using my device. I tried to clean the cache, logout/login again, HA restart, but no way to get an id created (on browser_mod integration page, I only have one device for my chrome on my laptop).

`browser_mod.debug` (launched from my android app) has no effect on my android app, while it creates a popup on my laptop browser (as expected).

**What I expected to happen:**
A popup comes

**What happened instead:**
Nothing

**Minimal** steps to reproduce:

```yaml
- type: entities
  entities:
    - entity: binary_sensor.meteoalarm
      tap_action:
        action: fire-dom-event
        browser_mod:
          command: popup
          deviceID:
            - this
          title: Alerte météo
          large: true
          card:
            type: markdown
            content: |
              {{'test'}}
```

**Error messages from the browser console:**
No error
```
```

---

**By replacing the space in the checkboxes ([ ]) with an X below, I indicate that I:**

- [x] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [x] Have made sure I am using the latest version of the plugin.

- [x] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [x] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by thomasgermain** [comment]
Ok so.. the device appears in the integration and popup (and I suppose everything else) is working well.

I don't know exactly what I did :nerd_face:, but I _think_ it started to work when I restarted my phone.

**Comment by thomasgermain** [comment]
Hmm Actually, the device is now unavailable and popups are not working anymore.

**Comment by Lampyridae1984** [comment]
I have the same issue :(

Tried to clear the cache on all devices and browsers connected/logged in to HA, but the issue remains. After activating the debug function, nothing happens - no popup and no new entities found in the integration.

**Comment by thomasgermain** [comment]
> I have the same issue :(
> 
> Tried to clear the cache on all devices and browsers connected/logged in to HA, but the issue remains. After activating the debug function, nothing happens - no popup and no new entities found in the integration.

I had to restart my android phone in order to the device in the integration. After that, it worked perfectly, but less then one day later, the device is unavailable and nothing is working on my android phone

**Comment by Lampyridae1984** [comment]
Unfortunately, that never worked for me.

**Comment by Lampyridae1984** [comment]
> > I have the same issue :(
> > Tried to clear the cache on all devices and browsers connected/logged in to HA, but the issue remains. After activating the debug function, nothing happens - no popup and no new entities found in the integration.
> 
> I had to restart my android phone in order to the device in the integration. After that, it worked perfectly, but less then one day later, the device is unavailable and nothing is working on my android phone

Unfortunately, that never worked for me.

**Comment by thomasgermain** [comment]
Here is my history, you can see sometimes the entity is available but I don't know why it becomes available and unavailable a bit later:
![image](https://user-images.githubusercontent.com/12560542/118937928-58014d00-b94e-11eb-8443-9eff7d78b19b.png)

**Comment by rnotley** [comment]
I had this same problem with my android tablet and was able to resolve it.  Go into your android's Settings app and navigate to:
**Privacy -> Ads**
I clicked **Reset advertising ID** as well as **Opt out of Ads Personalization**.  One of those did the trick and now I have a deviceID.

**Comment by Lampyridae1984** [comment]
> I had this same problem with my android tablet and was able to resolve it. Go into your android's Settings app and navigate to:
> **Privacy -> Ads**
> I clicked **Reset advertising ID** as well as **Opt out of Ads Personalization**. One of those did the trick and now I have a deviceID.

Unfortunately that didn't work either. :/

**Comment by dlevene1** [comment]
I'm using a browser_mod.popup with the following (or none of them as pretty sure defaults should be that)
```
          deviceID:
            - this
            - dashboard
```

But what's happening is the popup is appearing/running on my laptop. My phone(Home assistant app) and laptop are getting the same device ID.
I run from dev tools -> Services -> browser_mod.debug and see the same thing - the device ID pops up on my laptop.

I've also logged into the browser UI via my phone and it works ( and also opens the same on my latptop ). Seems the device ID for me is currently being applied account wide, when it needs to be device specific. Am I missing a setting/config someone to enable this?

**Comment by rnotley** [comment]
FWIW, I had a similar problem on my desktop where I use AdGuard.  I had to disable it to get a deviceID to show up.  Perhaps some other ad blocking or privacy service running on these devices is able to hide/mask/change the device id.  Just a thought, not sure how browser_mod gets the id.

**Comment by Lampyridae1984** [comment]
> I'm using a browser_mod.popup with the following (or none of them as pretty sure defaults should be that)
> 
> ```
>           deviceID:
>             - this
>             - dashboard
> ```
> 
> But what's happening is the popup is appearing/running on my laptop. My phone(Home assistant app) and laptop are getting the same device ID.
> I run from dev tools -> Services -> browser_mod.debug and see the same thing - the device ID pops up on my laptop.
> 
> I've also logged into the browser UI via my phone and it works ( and also opens the same on my latptop ). Seems the device ID for me is currently being applied account wide, when it needs to be device specific. Am I missing a setting/config someone to enable this?

Neither -This or -Dashboard work properly for, or perhaps I should say - they just don't. Does it work for anyone? BUT (!), I finally managed to get four working device IDs. I now got Android app ID, iOS app ID, Chrome on Windows ID and Chrome on Android ID.

To get the Android app ID assigned, I first reset the Android Ads settings mentioned above and then removed both the cache & stored data on both the Android app and Chrome for Android. (Not sure how much the reset of the Ad settings did, but still worth trying.) Then I just logged in again on the app and ran the debug function under Developer Tools. Voilà!!!

I'll keep you all posted on how long everything works as intended or if any new problems arise.

**Comment by Lampyridae1984** [comment]
> FWIW, I had a similar problem on my desktop where I use AdGuard. I had to disable it to get a deviceID to show up. Perhaps some other ad blocking or privacy service running on these devices is able to hide/mask/change the device id. Just a thought, not sure how browser_mod gets the id.

No type of AdBlocker is present on any of our browsers, neither desktop nor mobile. But who knows. In any case, nice that you got it to work! :)

**Comment by dlevene1** [comment]
Thanks for the tips - resetting app data and cache on the android app allowed it to start working as expected.

**Comment by LIRIKKER** [comment]
Hi, not sure whether to reopen this or create a new post...

But I've just installed Browser Mod through HACS. Devices seem to have different deviceID's. My instance of HASS has two profiles, me and my girlfriend. I first installed browser mod and used only my chrome instance where popups were working fine and everything was functioning as expected. I then restarted hass as I installed a new package to find that popups were no longer working. Nothing in the developer menu seemed to trigger anything either. I then opened the app on my android app and tried opening a popup to which is worked! On my PC it still didn't work. On my phone I went and used the debug command which showed my device ID. When on my computer I again ran the debug tool which showed a pop up on my android phone showing the same device ID. Very puzzled. I then asked my GF to open the app on her iPhone and see if Popups work. They did on her iPhone and her own profile but now have stopped working on my Android and Chrome. Running debug tool on my android and PC chrome seems to show a popup on now the iPhone on a different profile showing the iPhone's device ID which was different to the one on my android and chrome PC. 

Any idea as to what's going on?
